### PR TITLE
Fix typos in JdbcPagingItemReaderBuilder

### PR DESCRIPTION
**Pull request by drumonii**
Fix some spelling typos in javadoc, typo in an `IllegalArgumentException` message, and typo with an accidental field usage.

### CODE REVIEW

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/builder/JdbcPagingItemReaderBuilder.java [hunk 1]
```diff
@@ -331,7 +331,7 @@
 	private PagingQueryProvider determineQueryProvider(DataSource dataSource) {
 
 		try {
-			DatabaseType databaseType = DatabaseType.fromMetaData(this.dataSource);
+			DatabaseType databaseType = DatabaseType.fromMetaData(dataSource);
```
Comment (mminella) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/builder/JdbcPagingItemReaderBuilder.java]: The `this` should remain.  Using `this` is part of the Spring Framework coding standards: https://github.com/spring-projects/spring-framework/wiki/Code-Style#field-and-method-references

File: spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/builder/JdbcPagingItemReaderBuilder.java [hunk 2]
```diff
@@ -331,7 +331,7 @@
 	private PagingQueryProvider determineQueryProvider(DataSource dataSource) {
```
Comment (mminella) [inline spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/builder/JdbcPagingItemReaderBuilder.java]: This method could probably be static...

Comment (drumonii) [reply spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/builder/JdbcPagingItemReaderBuilder.java]: If I set that method to static then I'd need to pass in `selectClause`, `fromClause`, `whereClause`, `groupClause`, `sortKeys`, as they are referenced using `this`. But I guess at that point, might as well use `this.dataSource` without a parameter and keep it non static. Let me know...

Comment (mminella) [reply spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/builder/JdbcPagingItemReaderBuilder.java]: You're right.  That's what I get from reading just the diff.  Looks good to me. Thanks for the updates!

**Comment by benas** [comment]
LGTM. rebased and merged as a8ecb4fdcff643f4e71b9e11cb5df2a7452c5544.

Thank you @drumonii for your PR!

**Pull request closed without merging**